src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -269,6 +272,12 @@ public async Task SaveAsync()
                 this.pendingTipHashHeight = null;
             }
 
+            if (pendingBatch.Count == 0)
```

Comment:
Is there any chance that after `this.PendingBatch.Clear();`, `pendingBatch.Count != 0`?

Reply:
ah wait it's not the same object. Well that's a bit confusing lol.

Reply:
yeah it should be renamed I agree